### PR TITLE
feat: remove inactive pages from hmr

### PR DESCRIPTION
**Pull request by harrytothemoon**
**What kind of change does this PR introduce?**

Remove inactive page from HMR

**Did you add tests(unit,e2e) for your changes?**

no

**Does this PR introduce a breaking change?**

no

**What needs to be documented once your changes are merged?**

no

### CODE REVIEW

File: packages/service/src/server/middlewares/dev/hotMiddleware.ts [hunk 1]
```diff
@@ -125,17 +128,29 @@ export class WebpackHotMiddleware {
...
+      modulesActivity.set(pathname, {
+        absolutePath: route.__componentSourceWithAffix__!,
+        lastActivity: Date.now()
+      });
     }
   }
 
+  private handlePing(page: string | null): void {
+    if (!page || !modulesActivity.has(page)) return; //error page
+    modulesActivity.set(page, {
```
Comment (liximomo) [inline packages/service/src/server/middlewares/dev/hotMiddleware.ts:144]: 这个是不是和 updateModuleActivity 重复了？

File: packages/platform-web/src/shuvi-app/entry/client/app.ts [hunk 2]
```diff
@@ -34,6 +35,15 @@ const run = async () => {
 
 export { run };
 
+app.router.afterEach(() => {
```
Comment (liximomo) [inline packages/platform-web/src/shuvi-app/entry/client/app.ts:38]: 应该用 initHMR返回的devClient对象，封装性更好点，不要直接用 @shuvi/toolpack/lib/utils/hotDevClient/websocket 这个模块，

**Review by liximomo — CHANGES_REQUESTED** [review]

File: packages/service/src/server/middlewares/dev/hotMiddleware.ts [hunk 3]
```diff
@@ -101,13 +119,45 @@ export class WebpackHotMiddleware {
...
     this.closed = true;
     this.clientManager.close();
   };
+
+  private updateModuleActivity(matchRoutes: IRouteMatch[] | []): void {
+    if (matchRoutes.length < 1) return; //error page
+
+    for (const { route, pathname } of matchRoutes) {
+      modulesActivity.set(pathname, {
+        absolutePath: route.__componentSourceWithAffix__!,
```
Comment (liximomo) [inline packages/service/src/server/middlewares/dev/hotMiddleware.ts:136]: 这个不太对，应该用webpack的module id，__componentSourceWithAffix__并不是webpack的module id

**Review by liximomo — CHANGES_REQUESTED** [review]

File: packages/service/src/server/middlewares/dev/hotMiddleware.ts (hunk 1 above)
Comment (liximomo) [reply packages/service/src/server/middlewares/dev/hotMiddleware.ts:144]: setInterval(() => {
   notify(router.current)
}, 5000)

user
   detail

**Review by liximomo — CHANGES_REQUESTED** [review]
/lgtm

File: packages/service/src/server/middlewares/dev/hotMiddleware.ts (hunk 1 above)
Comment (harrytothemoon) [reply packages/service/src/server/middlewares/dev/hotMiddleware.ts]: 改動如下
https://github.com/shuvijs/shuvi/pull/382/commits/7451df7fcfe1b8e9fa6d2e641c57f8b2544b940f

**Pull request merged**